src/processor.ts

```diff
@@ -92,170 +118,191 @@ const processor = new SubstrateBatchProcessor()
+                            event: event,
                         })
                         break
                     case 'Farming.Withdrawn':
                         await handleFarmingWithdrawn({
                             ...ctx,
-                            block: block.header,
-                            event: item,
+                            block: block,
+                            event: event,
                         })
                         break
                     case 'Farming.Claimed':
                         await handleFarmingClaimed({
                             ...ctx,
-                            block: block.header,
-                            event: item,
+                            block: block,
+                            event: event,
                         })
                         break
                     case 'Farming.WithdrawClaimed':
                         await handleFarmingWithdrawClaimed({
                             ...ctx,
-                            block: block.header,
-                            event: item,
+                            block: block,
+                            event: event,
                         })
                         break
                     case 'Farming.GaugeWithdrawn':
                         await handleFarmingGaugeWithdrawn({
                             ...ctx,
-                            block: block.header,
-                            event: item,
+                            block: block,
+                            event: event,
                         })
                         break
                     // balances
                     case 'Balances.Transfer':
                         await handleBalanceTransfer({
                             ...ctx,
-                            block: block.header,
-                            event: item,
+                            block: block,
+                            event: event,
                         })
                         break
                     // contracts
                     case 'Contracts.ContractEmitted':
                         await handleContractEvent({
                             ...ctx,
-                            block: block.header,
-                            event: item,
+                            block: block,
+                            event: event,
                         })
                         break
                     // price oracle
                     case 'DiaOracleModule.UpdatedPrices':
                         await handleUpdatedPrices({
                             ...ctx,
-                            block: block.header,
-                            event: item,
+                            block: block,
+                            event: event,
                         })
                         break
                     default:
+                        await saveEvent(ctx, event)
```

Comment:
Wouldn't we want to save the event in any case, even if it was handled in one of the other cases? If so, we should move the call to save it out of the switch-case.